SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -0,0 +1,250 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Utils/InterestFactory.h"
+
+#include "Engine/World.h"
+#include "EngineClasses/SpatialNetConnection.h"
+#include "EngineClasses/SpatialNetDriver.h"
+#include "EngineClasses/SpatialPackageMapClient.h"
+#include "GameFramework/PlayerController.h"
+
+namespace improbable
+{
+
+InterestFactory::InterestFactory(AActor* InActor, const FClassInfo& InInfo, USpatialNetDriver* InNetDriver)
+	: Actor(InActor)
+	, Info(InInfo)
+	, NetDriver(InNetDriver)
+	, PackageMap(InNetDriver->PackageMap)
+{}
+
+Worker_ComponentData InterestFactory::CreateInterestData()
+{
+	return CreateInterest().CreateInterestData();
+}
+
+Worker_ComponentUpdate InterestFactory::CreateInterestUpdate()
+{
+	return CreateInterest().CreateInterestUpdate();
+}
+
+Interest InterestFactory::CreateInterest()
+{
+	if (Actor->IsA<APlayerController>())
+	{
+		return CreatePlayerControllerInterest();
+	}
+	else
+	{
+		return CreateActorInterest();
+	}
+}
+
+Interest InterestFactory::CreateActorInterest()
+{
+	Interest NewInterest;
+
+	QueryConstraint DefinedConstraints = CreateDefinedConstraints();
+
+	if (!DefinedConstraints.IsValid())
+	{
+		return NewInterest;
+	}
+
+	Query NewQuery;
+	NewQuery.Constraint = DefinedConstraints;
+	// TODO: Make result type handle components certain workers shouldn't see
```

Comment:
Can you make / attach a ticket to this TODO?

Reply:
I'm waiting for more people to review the Interest doc before I make tickets for it so I know it's the right direction.